src/ralph/admin/views/multiadd.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+from django import forms
+from django.conf.urls import url
+from django.contrib import messages
+from django.core.urlresolvers import reverse
+from django.db import IntegrityError, transaction
+from django.http import HttpResponseRedirect
+from django.shortcuts import get_object_or_404
+from django.utils.translation import ugettext_lazy as _
+
+from ralph.admin.fields import MultilineField, MultivalueFormMixin
+from ralph.admin.mixins import RalphTemplateView
+from ralph.admin.sites import ralph_site
+
+
+class MultiAddView(RalphTemplateView):
+
+    """Adding view to multiple adds models."""
+
+    template_name = 'admin/multi_add.html'
+
+    def dispatch(self, request, object_pk, model, *args, **kwargs):
+        admin_model = ralph_site._registry[model]
+        self.model = model
+        self.fields = admin_model.multiadd_fields
+        self.info_fields = admin_model.multiadd_info_fields
+        self.obj = get_object_or_404(model, pk=object_pk)
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_form(self):
+        form_kwargs = {}
+        multi_form_attrs = {
+            'multivalue_fields': self.fields,
+            'model': self.model
+        }
+        for field in self.fields:
+            multi_form_attrs[field] = MultilineField(allow_duplicates=False)
+
+        multi_form = type(
+            'MultiForm', (MultivalueFormMixin, forms.Form), multi_form_attrs
+        )
+        if self.request.method == 'POST':
+            form_kwargs['data'] = self.request.POST
+        return multi_form(**form_kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['form'] = self.get_form()
+        context['obj'] = self.obj
+        context['info_fields'] = self.info_fields
+        return context
+
+    def post(self, request, *args, **kwargs):
+        form = self.get_form()
+        if form.is_valid():
+            try:
+                return self.form_valid(form)
+            except IntegrityError as e:
+                context = self.get_context_data()
+                context['form'] = form
+                form.add_error(None, e.args[1])
+                return self.render_to_response(context)
+        else:
+            return self.form_invalid(form)
+
+    def form_invalid(self, form):
+        context = self.get_context_data()
+        context['form'] = form
+        return self.render_to_response(context)
+
+    def get_url_name(self):
+        """
+        Return URl for model change list.
+        """
+        params = self.model._meta.app_label, self.model._meta.model_name
+        url = 'admin:{}_{}_changelist'.format(*params)
+        return reverse(url)
+
+    @transaction.atomic
+    def form_valid(self, form):
+        saved_assets = []
+        args = [form.cleaned_data[field] for field in self.fields]
+        for data in zip(*args):
+            self.obj.baseobject_ptr_id = None
```

Comment:
as far as i understand, this view should be generic (easy to use for any model) - if yes, then we shouldn't rely here on asset/baseobject etc - find out if we can get inheritance hierarchy (and id's pointers) from meta or sth like that